### PR TITLE
Cleanup adding accounts and performance improvements

### PR DESCRIPTION
**Pull request by FrederikBolding**

### CODE REVIEW

**Comment by FrederikBolding** [comment]
Has the performance improvements of #3915 among other things.

**Comment by FrederikBolding** [comment]
[ch8342]

**Review by GuillaumeRx — APPROVED** [review]

File: src/services/Store/Contact/helpers.ts [hunk 1]
```diff
@@ -68,3 +77,32 @@ export const findMultipleNextUnusedDefaultLabels = (wallet: WalletId, numOfLabel
...
+  contacts: ExtendedContact[],
+  contracts: ExtendedContract[]
+) => (address: TAddress, networkId: NetworkId) => {
+  return (
+    [...contacts, ...STATIC_CONTACTS]
+      .filter((contact: ExtendedContact) => contact.network === networkId)
+      .find((contact: ExtendedContact) => isSameAddress(contact.address as TAddress, address)) ||
+    getContactFromContracts(contracts)(address, networkId)
+  );
+};
```
Comment (GuillaumeRx) [inline src/services/Store/Contact/helpers.ts:108]: Can't this be done in redux ?

File: src/services/Store/store/account.slice.ts [hunk 2]
```diff
@@ -216,6 +242,96 @@ export function* handleAddAccounts({ payload }: PayloadAction<IAccount[]>) {
...
+      }
+      return undefined;
+    })
+    .filter((a) => a !== undefined) as ExtendedContact[];
+  if (newContacts.length > 0) {
+    yield put(createOrUpdateContacts(newContacts));
+  }
+  yield put(addAccounts(newRawAccounts));
+  yield put(scanTokens({ accounts: newRawAccounts }));
+  yield put(fetchMemberships(newRawAccounts));
```
Comment (GuillaumeRx) [inline src/services/Store/store/account.slice.ts:313]: this could be done inside some `takeLatest` nah ?

**Review by GuillaumeRx — CHANGES_REQUESTED** [review]

File: src/services/Store/store/account.slice.ts (hunk 2 above)
Comment (FrederikBolding) [reply src/services/Store/store/account.slice.ts:313]: How? The format of the payload is not necessarily the same?

File: src/services/Store/Contact/helpers.ts (hunk 1 above)
Comment (FrederikBolding) [reply src/services/Store/Contact/helpers.ts:108]: Yes, but as discussed in DMs that would be out of scope for this PR.

File: src/services/Store/store/account.slice.ts (hunk 2 above)
Comment (GuillaumeRx) [reply src/services/Store/store/account.slice.ts:313]: Yeah I just saw that scanToken don't have the same payload

**Review by GuillaumeRx — APPROVED** [review]

**Pull request merged**